### PR TITLE
HIVE-28188:Upgrade Postgres to 42.7.3 to target CVE-2024-1597

### PR DESCRIPTION
**Pull request by devaspatikrishnatri**
### What changes were proposed in this pull request?
Upgrade Postgres to 42.5.6

### Why are the changes needed?
Target Critical CVE-2024-1597

### Does this PR introduce _any_ user-facing change?
No

### Is the change a dependency upgrade?
Yes
https://issues.apache.org/jira/secure/attachment/13068348/mvn_dependency_tree.txt

### How was this patch tested?
Locally Built relying on precommit tests

### CODE REVIEW

**Comment by zhangbutao** [comment]
Hi @devaspatikrishnatri, as this is a dependency upgrade,  you may need to provide the execution log of   `mvn dependency:tree`

**Comment by devaspatikrishnatri** [comment]
mvn_dependency_tree.txt

**Comment by devaspatikrishnatri** [comment]
@zhangbutao have attached the tree , Link https://issues.apache.org/jira/secure/attachment/13068348/mvn_dependency_tree.txt

File: standalone-metastore/pom.xml [hunk 1]
```diff
@@ -71,7 +71,7 @@
     <mariadb.version>2.5.0</mariadb.version>
     <mssql.version>6.2.1.jre8</mssql.version>
     <mysql.version>8.0.31</mysql.version>
-    <postgres.version>42.5.1</postgres.version>
+    <postgres.version>42.5.6</postgres.version>
```
Comment (zhangbutao) [inline standalone-metastore/pom.xml:74]: https://mvnrepository.com/artifact/org.postgresql/postgresql The latest version is 42.7.3, can we upgrade it to the latest version?

**Review by zhangbutao — APPROVED** [review]
+1 LGTM

**Comment by devaspatikrishnatri** [comment]
@zhangbutao Could you please merge it if this is okay..I do not have merge access yet in HIVE.

File: packaging/src/docker/README.md [hunk 2]
```diff
@@ -117,7 +117,7 @@ export POSTGRES_LOCAL_PATH=your_local_path_to_postgres_driver
 ```
 Example:
 ```shell
-mvn dependency:copy -Dartifact="org.postgresql:postgresql:42.5.1" && \
+mvn dependency:copy -Dartifact="org.postgresql:postgresql:42.7.3" && \
```
Comment (zhangbutao) [inline packaging/src/docker/README.md:120]: Could you also please fix the version in these lines? Thanks.
https://github.com/apache/hive/blob/1a969f6642d4081027040f97e6a689bf2e687db3/packaging/src/docker/README.md?plain=1#L63
https://github.com/apache/hive/blob/1a969f6642d4081027040f97e6a689bf2e687db3/packaging/src/docker/README.md?plain=1#L72
https://github.com/apache/hive/blob/1a969f6642d4081027040f97e6a689bf2e687db3/packaging/src/docker/README.md?plain=1#L121

**Pull request merged**